drivers/hsc/hsc.c

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (C) 2020 Deutsches Zentrum für Luft- und Raumfahrt e.V. (DLR)
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_hsc
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the Honeywell HSC
+ *              temperature and pressure sensor.
+ *
+ * @author      Quang Pham <phhr_quang@live.com>
+ *
+ * @}
+ */
+
+#include "hsc.h"
+#include "errno.h"
+#include "hsc_internals.h"
+#include "hsc_params.h"
+#include "periph/i2c.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+#define DEV_I2C      (dev->params.i2c_dev)
+#define DEV_ADDR     (dev->params.i2c_addr)
+
+/* Internal function prototypes */
+static int _read_ut(const hsc_t *dev, int32_t *ut);
+static int _read_up(const hsc_t *dev, int32_t *up);
+
+/*---------------------------------------------------------------------------*
+ *                          HSC Core API                                     *
+ *---------------------------------------------------------------------------*/
+
+int hsc_init(hsc_t *dev, const hsc_params_t *params)
+{
+    dev->params = *params;
+    uint8_t buf[HSC_FULL_DATA_LENGTH];
+
+    /* Acquire exclusive access */
+    if (i2c_acquire(DEV_I2C)) {
+        return -ENODEV;
+    }
+
+    if (i2c_read_bytes(DEV_I2C, DEV_ADDR, buf, sizeof(buf), 0) < 0) {
+        i2c_release(DEV_I2C);
+        DEBUG("[hsc] read_bytes fails\n");
+
+        return -EIO;
+    }
+
+    i2c_release(DEV_I2C);
+
+    /* Check the status of the sensor */
+    uint8_t status = buf[0];
+
+    switch (status & HSC_STATUS_MASK) {
+    case HSC_STATUS_OK:
+        return 0;
```

Comment:
```suggestion
        break;
```

The `return 0;` in line 73 is currently unreachable. You could either drop the `return 0;` there, or make it reachable by turning the `return 0;` here into a `break;`. I personally would prefer using the `break;` here, but I guess it is more a matter of personal taste.